### ISSUE TITLE
DataTypeInference's 'stripped' journal count is inflated by every missing value in the column

### ISSUE THREAD

**Issue by bijay-odyssey** [bijay-odyssey/edaprep#42]
`DataTypeInference`'s cast step counts how many values it whitespace-stripped, for the
journal:

```python
counts["stripped"] = int((stripped != series).sum())
```

`stripped != series` is an elementwise pandas comparison, and `NaN != NaN` evaluates
to `True` (not `NaN`, not `False`) -- so every missing value in the column counts as
"stripped" even though nothing was done to it.

### Reproduced

```python
df = pd.DataFrame({"s": ["a ", None, "b ", None, None]})  # 2 real strips, 3 NaN
DataTypeInference(["s"]).fit_transform(df, None, ctx)
# journal entry: {'s': {'stripped': 5}}   -- should be 2
```

For a column that's mostly missing, the reported "stripped" count is dominated by
phantom entries from rows nothing happened to. The transformed *values* are correct
(still `NaN`); only the journalled count is wrong, which matters here specifically
because "nothing is discarded/changed silently -- and correctly counted" is this
project's core guarantee, enforced via exactly this kind of journal entry.

### What to do

Exclude missing values from the comparison, e.g.
`int(((stripped != series) & series.notna()).sum())`, or compare only the non-null
subset.

### Tests

A column with a mix of strippable, clean, and missing values; assert
`journal.entries[...].effect["per_column"]["s"]["stripped"]` equals the count of
values that actually changed, not including `NaN` rows.

**Before you start:** please comment to claim this issue, and check whether someone
already has.

**Comment by alaa-bakr-analyst** [comment]
Hi! I'd like to claim this issue. I'll fix the stripped-value count so missing values aren't included, and add a regression test.

**Comment by bijay-odyssey** [comment]
Assigned — thanks for claiming it, and your scope is exactly right. Small and self-contained, as the label says.
